### PR TITLE
Adminhelps into mentorhelps converting.

### PR DESCRIPTION
**Pull request by AnywayFarus**
## About The Pull Request

This PR adds the ability to convert hits. From adminhelp to mentorhelp

## How This Contributes To The Skyrat Roleplay Experience

Easier way to send player to staff/mentors if they mistake admin/mentorhelp

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image

</details>

## Changelog

:cl: Vishenka0704
admin: Added a ticket into mentrohelp convertation. Better than say "write it to mentors".
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -210,6 +210,8 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	var/ticket_ping = FALSE
 	/// Who is handling this admin help?
 	var/handler
+	/// All sanitized text
+	var/full_text
```
Comment (GoldenAlpharex) [inline code/modules/admin/verbs/adminhelp.dm]: All of these Skyrat variables should be brought to a modular file instead.

File: code/modules/admin/verbs/adminhelp.dm [hunk 2]
```diff
@@ -399,6 +402,7 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=resolve'>RSLVE</A>)"
 	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=handle_issue'>HANDLE</A>)" //SKYRAT EDIT ADDITION
 	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=pingmute'>PING MUTE</A>)" //SKYRAT EDIT ADDITION
+	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=convert'>CONVERT</A>)" //SKYRAT EDIT ADDITION - Mentor/Admin Help Convertation
```
Comment (GoldenAlpharex) [inline code/modules/admin/verbs/adminhelp.dm]: Those two could probably be modularized easily with `. = ..()` in the modular version.

File: modular_skyrat/master_files/code/modules/admin/admin.dm [hunk 3]
```diff
@@ -45,3 +45,24 @@ GLOBAL_VAR_INIT(dchat_allowed, TRUE)
 
 	handler = "[usr.ckey]"
 	return TRUE
+
+///Procedure whick convert adminhelp to mentorhelp
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/admin/admin.dm]:
```suggestion
///Proc which converts an admin_help ticket to a mentorhelp
```

File: modular_skyrat/master_files/code/modules/admin/admin.dm [hunk 4]
```diff
@@ -45,3 +45,24 @@ GLOBAL_VAR_INIT(dchat_allowed, TRUE)
...
+/datum/admin_help/proc/convert(key_name = key_name_admin(usr))
+	if(state != AHELP_ACTIVE)
+		return FALSE
+	
+	if(handler && handler != usr.ckey)
+		var/response = tgui_alert(usr, "This ticket is already being handled by [handler]. Do you want to continue?", "Ticket already assigned", list("Yes", "No"))
+		if(!response || response == "No")
+			return FALSE
+	
+	add_verb(initiator, /client/verb/mentorhelp)
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/admin/admin.dm]: Why are you giving them the verb? I'm pretty sure everyone gets that verb by default, no?

File: modular_skyrat/master_files/code/modules/admin/admin.dm [hunk 5]
```diff
@@ -45,3 +45,24 @@ GLOBAL_VAR_INIT(dchat_allowed, TRUE)
 
 	handler = "[usr.ckey]"
 	return TRUE
+
+///Procedure whick convert adminhelp to mentorhelp
+/datum/admin_help/proc/convert(key_name = key_name_admin(usr))
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/admin/admin.dm]: Gotta give descriptive proc names.
```suggestion
/datum/admin_help/proc/convert_to_mentorhelp(key_name = key_name_admin(usr))
```

File: modular_skyrat/modules/mentor/code/mentorhelp.dm [hunk 6]
```diff
@@ -38,7 +38,7 @@
 		else
 			.["present"]++
 
-/proc/key_name_mentor(whom, include_link = null, include_name = FALSE, include_follow = FALSE, char_name_only = FALSE)
+/proc/key_name_mentor(whom, include_link = null, include_name = FALSE, include_follow = FALSE, char_name_only = FALSE, message)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/mentor/code/mentorhelp.dm]: That's just not how that's meant to be used.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'm fine with a way to convert from ahelp to mentor help, but I don't really think we need it the other way around. That's a lot of clutter for something that'd be very rarely used.

File: modular_skyrat/master_files/code/modules/admin/admin.dm (hunk 4 above)
Comment (AnywayFarus) [reply modular_skyrat/master_files/code/modules/admin/admin.dm]: if they have cooldown, nothing converts. if they dont have cooldown, mentorhelp dont duplicate. i think it's good solution

File: modular_skyrat/modules/mentor/code/mentorhelp.dm (hunk 6 above)
Comment (AnywayFarus) [reply modular_skyrat/modules/mentor/code/mentorhelp.dm]: without big re-working on mentor system it's easier way to get message, which used in this proc

Comment (AnywayFarus) [reply modular_skyrat/modules/mentor/code/mentorhelp.dm]: anyway.. removed

**Comment by AnywayFarus** [comment]
@GoldenAlpharex

**Comment by AnywayFarus** [comment]
не ври

**Comment by tf-4** [comment]
@GoldenAlpharex

**Comment by tf-4** [comment]
@GoldenAlpharex

**Comment by hugmeorelse** [comment]
I do like this simply because it encourages staff to send it to mentor-help so admin-help is reserved for the important tickets.

File: modular_skyrat/master_files/code/modules/admin/admin.dm [hunk 7]
```diff
@@ -19,6 +19,22 @@ GLOBAL_VAR_INIT(dchat_allowed, TRUE)
...
+	/// Who is handling this admin help?
+	var/handler
+	/// All sanitized text
+	var/full_text
+
+/datum/admin_help/ClosureLinks(ref_src)
+	. = ..()
+	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=handle_issue'>HANDLE</A>)" //SKYRAT EDIT ADDITION
+	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=pingmute'>PING MUTE</A>)" //SKYRAT EDIT ADDITION
+	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=convert'>CONVERT</A>)"
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/admin/admin.dm]: I feel like that would maybe be a bit more self-explaining than just "CONVERT".
```suggestion
	. += " (<A HREF='?_src_=holder;[HrefToken(forceGlobal = TRUE)];ahelp=[ref_src];ahelp_action=convert'>MHELP</A>)"
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Only critique and then we can give it a go.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me.

**Pull request merged**